### ISSUE TITLE
migration issue

### ISSUE THREAD

**Issue by mitologyllc** [stackblitz/bolt.new#4543]
### Describe the bug

I do not know coding so please bear with me. There is a migration problem, which I think Bolt.new has stuck in naming the new files. When it says it will create version 0024, the file name is still 0023. I have been in an indefinite loop of "applying changes" and "attempting to fix". I think 50% of my token is spent on this. Please advise. Thank you so very much!

### Link to the Bolt URL that caused the error

https://bolt.new/~/sb1-wete3ybd

### Validations

- [X] Please make your project public or accessible by URL. This will allow anyone trying to help you to easily reproduce the issue and provide assistance.

### Steps to reproduce

"Symptom Section":
1. Go to Symptom
2. Select any date/time
3. Select the severity of the symptoms.
Currently, it shows that the app is loading but the data cannot be saved. 

There are other issues I list below. Everytime I tried to fix it, the migration file naming issues show up. 
——————————————————————
 
*Test Results: 
1. Here is the current problem: (1) keep showing loading (2) input cannot be saved for standard tests (3) input cannot be saved for coustom tests. 
2. This is the feature I want for this section:
(1).    I would like to have a calendar, once data is input, there is a dot to show there is a log, and user can click the date to see a list down below shows what is recorded on that date.
(2).    Stanford Tests & Custom tests: the latest data should overwrite the previous input on the same date. For example, if I log in 3 different values on 12/20, I want the latest data overwrite previous log.

——————————————————————
 
*Diet Section:
1.    those dots shown in the calendar to indicate if there is data record or not, don’t show up sometimes.
2.    Data saving: Latest data on the same date is not saved. It seems to show up the first entry for that date.
3.    Need to make sure the Diet insights in Insight Section (Meal timing, nutrition) reflect the data in realtime.
 
—————————————————————— 

*Analytics Section:
⁃            For Symptoms, Is “Overall Trend” and “Average Severity” reflect the latest data log?
 
——————————————————————
 
*Insights Section:
1.    I want to make sure all the Summary Boxes (Symptom insights, Medication, Patterns, etc) reflect the data real-time.
2.    No réponse when I click individual insights box (Symptoms, Medication, etc) - lets remove these for now.
——————————————————————

 
*Dashboard Section:
1.    Is the “Today’s Symptoms” and “Medication Adherence” reflect the input data?
2.    There was a calendar to add next doctor appointments, which will also show up in “Reminders Section”, but now there is no place to add doc appointment.
3.    When I see if those updated in Reminder Section, the page is blank, the left hand side menu is gone too.
 
——————————————————————
 
*Profile:
1.    Selection is not saved.
2.    Add another input: Height
3.    In Analytic Section, under Weight Trend, add another BMI Chart (since now there is Height data)
 

### Expected behavior

"Symptom Section":
1. The selection should be reflected after click.
2. When user click on the calendar, the data should show correctly. If there is no log, the default is "no record"
3. Any date with logged data, there should be a dot marked in the calendar.

### Platform

```
Browser name = Chrome
Full version = 131.0.0.0
Major version = 131
navigator.appName = Netscape
navigator.userAgent = Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36
performance.memory = {
  "totalJSHeapSize": 288525289,
  "usedJSHeapSize": 239072205,
  "jsHeapSizeLimit": 4294705152
}
Username = mitologyllc
Chat ID = ed1740223148
```

### Additional context

THANK YOU.

**Comment by upsitesolutionsnl** [comment]
Same issue here, seems difficcult to resolve without resorting to manually influencing SQL or database.
![image](https://github.com/user-attachments/assets/95e2a2d8-40ee-44b2-81b7-c73502d5a9c4)
The chat prompt mentions creation a new version (06) for the migration, however it's still using an old migration number (02).

**Comment by mitologyllc** [comment]
@upsitesolutionsnl Yes. the same issue - in my case the file name number is always -1 of the version it says it will create/created.

**Comment by sadder76** [comment]
yes, i have the same and i ask Bolt to confirm it and it has been confirmed by Bolt :) 

![image](https://github.com/user-attachments/assets/e1d8026a-3746-4809-9754-61bbd7d73e6c)
![image](https://github.com/user-attachments/assets/37cf008d-1d99-4da0-be39-2710509c8eec)

anyway, Bolt create the following to describe the Bug:

# Bug Report: Migration File Version Conflicts

## Issue Description
The migration file versioning system is not properly handling version numbers, leading to duplicate version numbers being assigned to different migration files. This causes schema migration failures due to unique constraint violations on the schema_migrations_pkey.

## Steps to Reproduce
1. Create a new migration file
2. Create another migration file
3. Observe that both files may receive the same version number prefix (e.g., 0014_, 0015_)

## Current Behavior
Multiple migration files are being created with identical version numbers, for example:
- 0014_azure_shrine.sql
- 0014_bitter_star.sql
- 0014_still_water.sql

## Expected Behavior
Each new migration file should receive a unique, incremented version number to maintain proper migration sequence and avoid conflicts.

## Impact
- Schema migrations fail with error: `duplicate key value violates unique constraint "schema_migrations_pkey"`
- Blocks database schema updates
- Requires manual intervention to fix version numbers

## Technical Details
Error message:
```sql
ERROR: 23505: duplicate key value violates unique constraint "schema_migrations_pkey"
DETAIL: Key (version)=($VERSION) already exists.
